utoipa-swagger-ui/build.rs

```diff
@@ -32,10 +33,18 @@ fn main() {
     let zip_path = [&target_dir, &zip_filename].iter().collect::<PathBuf>();
 
     if !zip_path.exists() {
-        println!("start download to : {:?}", zip_path);
-        download_file(&url, zip_path.clone()).unwrap();
+        if url.starts_with("http://") || url.starts_with("https://") {
+            println!("start download to : {:?}", zip_path);
+            download_file(&url, zip_path.clone()).unwrap();
+        } else if url.starts_with("file://") {
+            let file_path = url.replace("file://", "");
```

Comment:
Arguably, file URLs should still be parsed as URLs. If it looks like a URL I'd usually assume urlencoding and such to apply as usual (on the other hand, building the path becomes easier when you just have to prepend file://... sigh).

Reply:
Except for space in a path, any other case where prepending `file://` is different from parsing as url ?

Reply:
Lots of special characters end up percentage-encoded. The most obvious case would be that `%` is used as the escape character, so it itself needs to be escaped as `%25`.

Reply:
Good point, though, not sure whether this actually comes to a problem, but if it does, I guess this could then be addressed. While I did not find a way to simply url encode string for the purpose with a quick look I assume as the `ZipArchive` works with rust `File` which itself uses `Path` to describe the location of a file system agnostic manner it should just simply fail if for one reason or another the rust cannot interpret the input as a valid `OsStr` that points to a real location.

That said e.g. Unix BASH does require strings to be double quoted to prevent the word globbing thus the variable could just be defined as `SWAGGER_UI_DOWNLOAD_URL="/path/to/my spaced swagger ui.zip"` if I am not mistaken.

Reply:
True.. Not a bad idea, I guess the `Url` is from the reqwest crate?

Reply:
I was thinking of crate [url](https://crates.io/crates/url) and method [to_file_path](https://docs.rs/url/2.2.2/url/struct.Url.html#method.to_file_path).
What u think ?

Reply:
Seems okay, I checked the amount of dependencies and it only has few of them so think we can have this there as well. I believe it does not increase the compile time enormously to out weight the benefit of "validating" URL.

Reply:
I need to double check, but I think reqwest::Url is already just a reexport of url::Url.

Reply:
Indeed `reqwest` can be used.  
So I committed to my fork: 
```rs
let file_path = Url::parse(&url).unwrap().to_file_path().unwrap();
```
Hopefully that should solve this edge case.

Reply:
Since the current PR is merged already, I created a new one with just this improvement: https://github.com/juhaku/utoipa/pull/925

Reply:
Merged that one as well